sdk/go/src/sawtooth_sdk/client/signer.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	ellcurv "github.com/btcsuite/btcd/btcec"
+	"math/big"
+)
+
+// -- Key Gen --
+
+// GenPrivKey generates a new private key and returns it as bytes
+func GenPrivKey() []byte {
+	priv, err := ellcurv.NewPrivateKey(ellcurv.S256())
+	if err != nil {
+		panic("Failed to generate private key")
+	}
+	return priv.Serialize()
+}
+
+// GenPubKey generates a new public key from a given private key and returns it
+// using the 33 byte compressed format
+func GenPubKey(privkey []byte) []byte {
+	_, pub := ellcurv.PrivKeyFromBytes(ellcurv.S256(), privkey)
+	return pub.SerializeCompressed()
+}
+
+// -- Signing --
+
+// Sign uses the given private key to calculate a signature for the given data.
+// A sha256 hash of the data is first calculated and this is what is actually
+// signed. Returns the signature as bytes using the compact serialization
+// (which is just (r, s)).
+func Sign(data, privkey []byte) []byte {
+	priv, _ := ellcurv.PrivKeyFromBytes(ellcurv.S256(), privkey)
+
+	hash := SHA256(data)
+
+	sig, err := priv.Sign(hash)
+	if err != nil {
+		panic("Signing failed")
+	}
+
+	return serializeCompact(sig)
+}
+
+// Verify uses the given public key to verify that the given signature was
+// created from the given data using the associated private key. A sha256 hash
+// of the data is calculated first and this is what is actually used to verify
+// the signature.
+func Verify(data, signature, pubkey []byte) bool {
+	sig := deserializeCompact(signature)
+	hash := SHA256(data)
+	pub, err := ellcurv.ParsePubKey(pubkey, ellcurv.S256())
+	if err != nil {
+		panic(err.Error())
+	}
+	return sig.Verify(hash, pub)
+}
+
+// -- SHA --
+
+// SHA512 calculates a sha512 has from the given input byte slice
+func SHA512(input []byte) []byte {
+	hash := sha512.New()
+	hash.Write(input)
+	return hash.Sum(nil)
+}
+
+// SHA256 calculates a sha256 has from the given input byte slice
+func SHA256(input []byte) []byte {
+	hash := sha256.New()
+	hash.Write(input)
+	return hash.Sum(nil)
+}
+
+// ---
+
+func serializeCompact(sig *ellcurv.Signature) []byte {
```

Comment:
Different casing than other functions? Not sure if typo or I just don't know Go styling.